js/onoffice-default-form-values.js

```diff
@@ -191,12 +191,15 @@ onOffice.default_values_input_converter = function () {
 
         if (!fieldDefinition.rangefield &&
             [
-                'integer', 'float', 'date',
+                'integer', 'float', 'date', 'datetime',
                 'urn:onoffice-de-ns:smart:2.5:dbAccess:dataType:float',
                 'urn:onoffice-de-ns:smart:2.5:dbAccess:dataType:int',
                 'urn:onoffice-de-ns:smart:2.5:dbAccess:dataType:integer',
                 'urn:onoffice-de-ns:smart:2.5:dbAccess:dataType:decimal'
             ].indexOf(fieldDefinition.type) >= 0) {
+            if (fieldDefinition.type === 'date' || fieldDefinition.type === 'datetime') {
```

Comment:
Is this check redundant with the previous check?

Reply:
I just though that

```php
[/*...*/ 'date', 'datetime' /*...*/].indexOf(fieldDefinition.type) >= 0
```

is the same as

```php
fieldDefinition.type === 'date' || fieldDefinition.type === 'datetime'
```
But I leave it up to you. :)

Reply:
If you use the previous check, the input text will be apply for the integer type also. That why we need another condition to render the correct input type.

Reply:
Oh, sorry! Now I see it. For some reason I didn't see that the inner if only covers one line and that there are more after it! 🤦